ethcore/src/engines/block_reward.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use ethabi;
+use ethabi::ParamType;
+use ethereum_types::{H160, Address, U256};
+
+use block::ExecutedBlock;
+use error::Error;
+use machine::EthereumMachine;
+use super::SystemCall;
+
+use_contract!(block_reward_contract, "BlockReward", "res/contracts/block_reward.json");
+
+/// The kind of block reward.
+/// Depending on the consensus engine the allocated block reward might have
+/// different semantics which could lead e.g. to different reward values.
+#[repr(u8)]
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum RewardKind {
```

Comment:
We already have a `RewardType` which is used for tracing but it only distinguishes between `Block` and `Uncle` rewards.